backend/db/analysis.py

```diff
@@ -165,23 +165,15 @@ def counts_over_time(
 
     # Pivot
     values = 'count' if method == 'counts_table' else 'delta'
-    data_df = current_counts_df.pivot(index='table', columns='timestamp', values=values).fillna(0).astype(int)
+    df = current_counts_df.pivot(index='table', columns='timestamp', values=values).fillna(0).astype(int)
 
     # Add note
     with get_db_connection(schema='', local=local) as con:
         runs = [dict(x) for x in sql_query(
             con, f"SELECT timestamp, note FROM counts_runs WHERE schema = '{schema}';", return_with_keys=True)]
     timestamps = [x['timestamp'] for x in runs]
-    ts_dict = {}
-    for ts in timestamps:
-        for run in runs:
-            if run['timestamp'] == ts:
-                ts_dict[ts] = run['note']
-    runs_df = pd.DataFrame([ts_dict])
-    runs_df = runs_df.reindex(sorted(runs_df.columns), axis=1)
-    runs_df.index = ['COMMENT']
-    df = pd.concat([runs_df, data_df])
-
+    runs_df = pd.DataFrame([timestamps])
+    df.columns = runs_df.iloc[0].tolist()
```

Comment:
Basically the issue is that the timestamps have more entries than the data columns. I looked at the SQL table that is producing the timestamps and there have been a bunch of failed refreshes that have their own timestamps. I haven't been able to look at the table that produces the df (it will take me some time to figure out how to do that because I'm not too familiar with the code and SQL etc) but I think that the failed refreshes are columns that are not showing up in the df variable. This is quite interesting and we might need a different approach for adding entries to the counts_runs table, I'll have to think about this more because I think the changes I made to the variables for db_refresh is creating this issue.

Reply:
@hugwuoke Oh, I see. Good investigation! I really appreciate you taking the time to think ahead and put your reasoning here.

That definitely sounds plausible. And that sounds potentially tricky as well. I'm trying to think as well what would make the number of columns different than the number of timestamps. They should be the same, but I would not be surprised if they somehow were not either. 

For the moment, I'll leave it to you (or Siggie) to prioritize where this should go in your list of priorities. The easiest way for me to help with your priorities would be through weekly/biweekly meetings (1:1 or the whole TermHub team).